app/controllers/file_manager_controller.rb

```diff
@@ -0,0 +1,228 @@
+# Code for file manager adapted from: adrientoub/file-explorer
+require 'archive'
+require 'pathname'
+
+class FileManagerController < ApplicationController
+  BASE_DIRECTORY = Rails.root.join('courses')
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :update_persistent_announcements
+
+  def index
+    path = params[:path].nil? ? "" : params[:path]
+    absolute_path = check_path_exist(path)
+    if (File.directory?(absolute_path) && check_instructor(absolute_path) ) || path == ""
+      populate_directory(absolute_path, "#{path}/")
+      render 'file_manager/index'
+    elsif File.file?(absolute_path) && check_instructor(absolute_path)
+      if File.size(absolute_path) > 1_000_000 || params[:download]
+        send_file absolute_path
+      else
+        @file = File.read(absolute_path)
```

Comment:
Nit: I think `File.read` technically requires a string while `absolute_path` is a pathname. In fact, you could do `absolute_path.read` instead

Reply:
Fixed.